packages/protocol/test-sol/integration/Integration.t.sol

```diff
@@ -1,19 +1,81 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.5.13;
+pragma solidity >=0.5.13 <0.8.20;
 
 import "celo-foundry/Test.sol";
-import "@celo-contracts/common/GoldToken.sol";
-// import "@celo-contracts/common/test/MockGoldToken.sol";
-import "forge-std/console.sol";
+
+import { Utils } from "@test-sol/utils.sol";
+import { Constants } from "@test-sol/constants.sol";
+
 import "@celo-contracts/common/interfaces/IRegistry.sol";
+import "@celo-contracts/common/interfaces/IProxy.sol";
 
 contract IntegrationTest is Test {
   address constant registryAddress = address(0x000000000000000000000000000000000000ce10);
-  address account1 = actor("account1");
-  address account2 = actor("account2");
   IRegistry registry = IRegistry(registryAddress);
 
   function setUp() public {}
+}
+
+contract RegistryIntegrationTest is IntegrationTest, Utils, Constants {
+  IProxy proxy;
+
+  function test_shouldHaveAddressInRegistry() public view {
+    for (uint256 i = 0; i < contractsInRegistry.length; i++) {
+      string memory contractName = contractsInRegistry[i];
+      address contractAddress = registry.getAddressFor(keccak256(abi.encodePacked(contractName)));
+      console2.log(contractName, "address in Registry is: ", contractAddress);
+      assert(contractAddress != address(0));
+    }
+  }
+
+  function test_shouldHaveCorrectBytecode() public {
+    for (uint256 i = 0; i < contractsInRegistry.length; i++) {
+      // Read name from list of core contracts
+      string memory contractName = contractsInRegistry[i];
+      console2.log("Checking bytecode of:", contractName);
+
+      // Converting contract names to hashes for comparison
+      bytes32 hashContractName = keccak256(abi.encodePacked(contractName));
+      bytes32 hashAccount = keccak256(abi.encodePacked("Accounts"));
+      bytes32 hashElection = keccak256(abi.encodePacked("Election"));
+      bytes32 hashEscrow = keccak256(abi.encodePacked("Escrow"));
+      bytes32 hashFederatedAttestations = keccak256(abi.encodePacked("FederatedAttestations"));
+      bytes32 hashGovernance = keccak256(abi.encodePacked("Governance"));
+      bytes32 hashSortedOracles = keccak256(abi.encodePacked("SortedOracles"));
+      bytes32 hashValidators = keccak256(abi.encodePacked("Validators"));
+
+      // Skipping test for contracts that depend on linked libraries
+      // This is a known limitation in Foundry at the moment:
+      // Source: https://github.com/foundry-rs/foundry/issues/6120
```

Comment:
looks like this issue was fixed in  `nightly-fbad377ab26a432e48444cf324feee1195a30960`

Reply:
Great point, they merged a PR that fixes this:
- Issue: https://github.com/foundry-rs/foundry/issues/6120
- PR: https://github.com/foundry-rs/foundry/pull/8034

Unfortunately, I can't use the latest version, because of a regression it has in `getDeployedCode`, which is a known issue: 
- https://github.com/smartcontractkit/foundry-chainlink-toolkit/issues/15

For the time being, I had to downgrade to last version that doesn't have the regression.

Reply:
I'll keep this comment unresolved, so it can be found for future reference.